lib/rules/destructuring-assignment.js

```diff
@@ -247,7 +248,8 @@ module.exports = {
           && destructureInSignature === 'always'
           && node.init.name === 'props'
         ) {
-          const scopeSetProps = context.getScope().set.get('props');
+          const scope = astUtil.getScope(context, node);
```

Comment:
will this behave differently, since it wasn't passing the node before?

Reply:
1. I just _want to believe_ it works similar enough as far [the migration guide](https://eslint.org/docs/latest/use/migrate-to-9.0.0) suggest this replacement.
2. I can rearrange the commits stack later, but I want to fix the tests first.

Reply:
The worst part: 
> Additionally, methods Linter#defineRule(), Linter#defineRules(), Linter#defineParser(), and Linter#getRules() are no longer available.